src/main/kotlin/uk/gov/justice/digital/hmpps/nomisuserrolesapi/config/NomisUserRolesApiExceptionHandler.kt

```diff
@@ -62,7 +64,8 @@ class NomisUserRolesApiExceptionHandler {
         ErrorResponse(
           status = BAD_REQUEST,
           userMessage = "Validation failure: ${e.message}",
-          developerMessage = e.message
+          developerMessage = e.message,
+          errorCode = BASIC_VALIDATION_FAILURE
         )
```

Comment:
For the various password validation errors now return a different error code so the client can tell the differences without relying on the message text